libs/blocks/preflight/checks/assets.js

```diff
@@ -19,9 +19,11 @@ function loadVideo(asset) {
     && asset.videoWidth > 0) return Promise.resolve();
 
   return new Promise((resolve) => {
-    asset.appendChild(createTag('source', { src: asset.getAttribute('data-video-source'), type: 'video/mp4' }));
-    ['loadedmetadata', 'error'].forEach((evt) => asset.addEventListener(evt, resolve, { once: true }));
-    asset.load();
+    if (!asset.querySelector('source')) {
+      asset.appendChild(createTag('source', { src: asset.getAttribute('data-video-source'), type: 'video/mp4' }));
+      ['loadedmetadata', 'error'].forEach((evt) => asset.addEventListener(evt, resolve, { once: true }));
+      asset.load();
+    }
```

Comment:
Is it possible that this Promise hangs ? If `asset` has `source` but `readyState` and `videoWidth` checks fail, Promise will be returned that never resolves right ? I would say event listeners should be moved outside if.
```suggestion
    if (!asset.querySelector('source')) {
      asset.appendChild(createTag('source', { src: asset.getAttribute('data-video-source'), type: 'video/mp4' }));
    }
    ['loadedmetadata', 'error'].forEach((evt) => asset.addEventListener(evt, resolve, { once: true }));
    asset.load();
```